### PR TITLE
Add spidermonkey report

### PR DESCRIPTION
**Pull request by gterzian**

### CODE REVIEW

File: _posts/2024-04-05-spidermonkey.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+- [Modular JS/execution engine](https://github.com/servo/servo/issues/30863)
+- [WebIDL impl: Replace use of NonNull<JSObject>](https://github.com/servo/servo/issues/30889)
+- [WebIDL impl: remove unsafe JSObject when returning a ReadableStream](https://github.com/servo/servo/issues/30891)
+- [WebIDL impl: remove unsafe JSObject from WebGLExtensionWrapper](https://github.com/servo/servo/issues/30892)
+- [Support FinalizationRegistry](https://github.com/servo/servo/issues/31072)
+
+--------
+
+Thank you for reading. This report was written by [Gregory Terzian](https://github.com/gterzian). Questions and comments can be made by opening an issue at [the original repo](https://github.com/gterzian/spidermonkey_servo), or by [reaching out on Zulip](https://servo.zulipchat.com).
```
Comment (mrego) [inline _posts/2024-04-05-spidermonkey.md:98]: About authorship, blog posts are from Servo as a whole, so won't say anything on the blog post itself. On the report, you can write the author and date (similar to what we have done in the last report: https://github.com/servo/servo/wiki/Servo-Layout-Engines-Report).

I think for questions, it'd be better to fill issues on Servo's GitHub directly.

**Review by mrego — COMMENTED** [review]
Is this a blog post about the report or the report itself? It looks like the later.

I think it'd be nicer to write a blog post about the report, a high level overview of the problems and solutions we're looking into, and link to the actual report that should be in the Servo's wiki.

File: _posts/2024-04-05-spidermonkey.md (hunk 1 above)
Comment (gterzian) [reply _posts/2024-04-05-spidermonkey.md:98]: Ok, thanks for the info, have added the report to https://github.com/servo/servo/wiki/Servo-and-SpiderMonkey-Report, and updated the link at https://github.com/servo/servo/wiki/Design#script

Will turn this into a short blog post.

Comment (mrego) [reply _posts/2024-04-05-spidermonkey.md:98]: Cool, thanks!

**Review by delan — APPROVED** [review]
Thanks! I’ve done some minor copyediting [in this post](https://github.com/servo/servo.org/compare/aaad32523c5dfe800ff1f7025f506bcd3dc2cc76...839603a49befd84f3917655ef9dc67c97d937ec5) and [in the full report](https://github.com/servo/servo/wiki/Servo-and-SpiderMonkey-Report/_compare/d84ac58e6569f68b55333bbf36942aad58046a8e...2bc57313179dbc5fbb26538d74c163e15142e0b2).

**Comment by gterzian** [comment]
@delan Thanks for the editing!

**Pull request merged**